File: src/main/java/dev/paigewatson/layoutmaster/client/services/FreightCarService.java
package dev.paigewatson.layoutmaster.client.services;

import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation;
import dev.paigewatson.layoutmaster.models.rollingstock.RollingStock;

import java.util.List;

public interface FreightCarService
{
    List<RollingStock> allFreightCars();

    List<RollingStock> allFreightCarsByAARType(AARDesignation aarDesignation);

    void delete(RollingStock rollingStockToDelete);

    List<RollingStock> allFreightCarsByRoadName(String roadName);

    RollingStock saveFreightCarToDatabase(RollingStock freightCarToSave);
}

File: src/main/kotlin/dev/paigewatson/layoutmaster/client/services/FreightCarService.kt
package dev.paigewatson.layoutmaster.client.services

import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation
import dev.paigewatson.layoutmaster.models.rollingstock.RollingStock

interface FreightCarService {
    fun allFreightCars(): List<RollingStock>
    fun allFreightCarsByAARType(aarDesignation: AARDesignation): List<RollingStock>
    fun delete(rollingStockToDelete: RollingStock)
    fun allFreightCarsByRoadName(roadName: String): List<RollingStock>
    fun saveFreightCarToDatabase(freightCarToSave: RollingStock): RollingStock
}
